src/json/mapping.cr

```diff
@@ -78,7 +78,7 @@ module JSON
         {% for key, value in properties %}
           when {{value[:key] || key.id.stringify}}
             %var{key.id} =
-            {% if value[:nilable] == true %} %pull.read_null_or { {% end %}
+            {% if value[:nilable] == true || value[:default] != nil %} %pull.read_null_or { {% end %}
```

Comment:
It took me a while to understand why we need the `value[:default] != nil` check here: it will set the default value if an explicit `null` comes in the JSON. I think a spec is missing for this (I removed that check and tests still pass). Could you add a spec for it?